### DISCUSSION TITLE
Problem regarding on undoing after getting extralife.

### DISCUSSION THREAD

**Discussion by hontszpang** [CastleLab/COMP3021-2021Fall-PA1-Student-Version#24] [Q&A] [answered]
For the Obfuscated JAR. Initially, if the player has 1 life, after that he makes a move and picks up an extra life and then does another move that hits the mine and losses 1 life. After that, the player keeps undoing until there is no move to undo. Then the player gets 0 life but the game still does not end.  Then the player with 0 lives makes a move and hits a mine, a runtime error results. Should we consider this kind of case?

**Comment by Derppening** [comment, marked as answer]
Are you using the old obfuscated JAR or the new one? I think this issue should be addressed in the updated obfuscated JAR in 29d891b (#23).

If it has not been addressed by the latest obfuscated JAR, could you send a sample map and the input sequence which triggers this buggy condition?

**Reply by hontszpang** [reply]
I think I am using the new one. I am using the attached map below with input sequence: right left down undo undo down. Thx!
[sample.txt](https://github.com/CastleLab/COMP3021-2021Fall-PA1-Student-Version/files/7250304/sample.txt)

**Reply by Derppening** [reply]
I see. It does appear to be a bug in my reference implementation.

Don't worry, I will not test this behavior, so you are free to handle this however you want.
